plugins/performance-lab/includes/admin/load.php

```diff
@@ -213,6 +213,46 @@ function perflab_dismiss_wp_pointer_wrapper(): void {
 }
 add_action( 'wp_ajax_dismiss-wp-pointer', 'perflab_dismiss_wp_pointer_wrapper', 0 );
 
+/**
+ * Gets the path to a script or stylesheet.
+ *
+ * @since n.e.x.t
+ *
+ * @param string      $src_path Source path.
+ * @param string|null $min_path Minified path. If not supplied, then '.min' is injected before the file extension in the source path.
+ * @return string URL to script or stylesheet.
+ */
+function perflab_get_asset_path( string $src_path, ?string $min_path = null ): string {
+	if ( null === $min_path ) {
+		// Note: wp_scripts_get_suffix() is not used here because we need access to both the source and minified paths.
+		$min_path = (string) preg_replace( '/(?=\.\w+$)/', '.min', $src_path );
+	}
+
+	$force_src = false;
+	if (
+		( WP_DEBUG || wp_get_environment_type() === 'local' )
```

Comment:
Makes sense to use `WP_DEBUG` around this check, because a production build should never be subject to this check.

I don't see that being used around the checks in our other plugins like Embed Optimizer, Image Prioritizer, and Optimization Detective though. Those should also check this only if `WP_DEBUG` is enabled. I understand for those having a non-built access is more critical, but that matters just as little for production builds of them as it does for a production build of PL here.

Reply:
I included the `'local'` check because just because you're on a non-production environment doesn't mean that you're going to have `WP_DEBUG` enabled. Sites spun up with Local do not have `WP_DEBUG` enabled. Also, even wp-env does not have `WP_DEBUG` enabled in the test environment (see #1271). So I was intending to increase the opportunities for the issue to be detected. That said, if `WP_DEBUG` is not enabled, then `wp_trigger_error()` doesn't do anything, so the result would be that the minified script would continue to be served even though it would result in a 404 (e.g. on production).

The `'local'` check isn't in the other plugins I was intending to roll out this check to the other plugins once reviewed here. But I'm less convinced now. I'll remove the `'local'` env check.

Reply:
My point in this comment was mostly that I think we should also add a `WP_DEBUG` check before the checks in those plugins, because it shouldn't be needed for production builds of the plugin anyway. Something for another PR of course, but I think that would be good to add.

It's not only about whether `wp_trigger_error()` actually triggers the error, but also about the check itself. Since it does nothing when `WP_DEBUG` is disabled, that's arguably an even better reason to wrap the check with `WP_DEBUG` as well.